src/runtime/pythonexception.cs

```diff
@@ -45,14 +45,18 @@ public PythonException()
             }
             if (_pyTB != IntPtr.Zero)
             {
-                using (PyObject tb_module = PythonEngine.ImportModule("traceback"))
-                {
-                    Runtime.XIncref(_pyTB);
-                    using (var pyTB = new PyObject(_pyTB))
-                    {
-                        _tb = tb_module.InvokeMethod("format_tb", pyTB).ToString();
-                    }
+                using PyObject tb_module = PythonEngine.ImportModule("traceback");
+
+                Runtime.XIncref(_pyTB);
+                using var pyTB = new PyObject(_pyTB);
+
+                using var tbList = tb_module.InvokeMethod("format_tb", pyTB);
+
+                var sb = new StringBuilder();
+                foreach (var line in tbList) {
+                    sb.Append(line.ToString());
                 }
+                _tb = sb.ToString();
```

Comment:
I vaguely remember you need to reverse the python stack trace lines int order for them to match the order in `base.StackTrace`. Can you please doublecheck.

Reply:
You are right. Python tracebacks are sorted from outer to inner functions, C# are sorted from inner to outer. Since we retain the respective formatting, I wonder whether it's really a good idea to turn own of them (likely the Python one) around as that may mess with expectations of the debugging user...

Reply:
I did so in my fork (e.g. reversed `tbList`), because it was really confusing during debugging.